watertap/unit_models/nanofiltration_0D.py

```diff
@@ -0,0 +1,675 @@
+            self.scale_constraint_by_nominal_value(
+                c,
+                scheme=ConstraintScalingScheme.inverseMaximum,
+                overwrite=overwrite,
+            )
+
+
+@declare_process_block_class("Nanofiltration0D")
+class Nanofiltration0DData(UnitModelBlockData):
+    """
+    Zero order nanofiltration model.
+    """
+
+    default_initializer = Nanofiltration0DInitializer
+    default_scaler = Nanofiltration0DScaler
+
+    CONFIG = ConfigDict()
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. Product blocks are always steady-state.""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Product blocks do not contain holdup, thus this must be
+    False.""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for mixer",
+            doc="""Property parameter object used to define property
+    calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigDict with arguments to be passed to a property
+    block(s) and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "include_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include pressure balance for retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "has_retentate_pressure_drop",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            description="Whether to include pressure drop in the retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "include_temperature_equality",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include temperature equality constraints",
+            doc="""Argument indicating whether temperature equality should be
+            inlcluded. If this is False, no energy balance constraint will be
+            written.""",
+        ),
```

Comment:
I did debate this whilst writing the model, and decided to not be overly constrained by standards and instead focus on what made sense for the case at hand. Config option names are easily changed at least, but my initial feeling is that a control volume is unnecessary complications in this case.

Reply:
If we didn't already have the NF_ZO model, then I would agree. Since we do though, I think the merging of the two should resolve this, and we can just keep the "standard" config options from the ZO formulation while adding your electroneutrality option.